src/storage/backend/sqlite/SQLite.cc

```diff
@@ -2,13 +2,58 @@
 
 #include "zeek/storage/backend/sqlite/SQLite.h"
 
+#include <thread>
+
 #include "zeek/3rdparty/sqlite3.h"
 #include "zeek/Func.h"
 #include "zeek/Val.h"
 #include "zeek/storage/ReturnCode.h"
 
+using namespace std::chrono_literals;
+
 namespace zeek::storage::backend::sqlite {
 
+OperationResult SQLite::RunPragma(std::string_view name, std::optional<std::string_view> value) {
+    int attempts = 0;
+    char* errorMsg = nullptr;
+    std::chrono::milliseconds time_spent = 0ms;
+    std::chrono::milliseconds backoff = 1ms;
+
+    std::string cmd = util::fmt("pragma %.*s", static_cast<int>(name.size()), name.data());
+    if ( value && ! value->empty() )
+        cmd += util::fmt(" = %.*s", static_cast<int>(value->size()), value->data());
+
+    while ( pragma_timeout > 0ms && time_spent < pragma_timeout ) {
+        int res = sqlite3_exec(db, cmd.c_str(), NULL, NULL, &errorMsg);
+        if ( res == SQLITE_OK ) {
+            break;
+        }
+        else if ( res == SQLITE_BUSY ) {
+            printf("busy?\n");
+            // If we got back that the database is busy, it likely means that another process is trying to
+            // do their pragmas at startup too. Exponentially back off and try again after a sleep.
+            sqlite3_free(errorMsg);
+            std::this_thread::sleep_for(backoff);
+            time_spent += backoff;
+            backoff *= 10;
+        }
+        else {
+            std::string err =
+                util::fmt("Error while executing %.*s pragma", static_cast<int>(name.size()), name.data());
+            sqlite3_free(errorMsg);
+            return {ReturnCode::INITIALIZATION_FAILED, std::move(err)};
+        }
+    }
+
+    if ( time_spent >= pragma_timeout ) {
+        std::string err =
+            util::fmt("Database was busy while executing %.*s pragma", static_cast<int>(name.size()), name.data());
```

Comment:
```suggestion
            util::fmt("Database was busy while executing pragma '%s'", cmd.c_str());
```

Would that be okay, also above - it might have interesting information.